18 Normalizr/src/pages/student-detail/components/studentForm.tsx

```diff
@@ -1,13 +1,15 @@
-import * as React from "react";
-import { Input } from "../../../common/components/Input";
-import { StudentView } from "../../../model/view/studentView";
-import { StudentErrors } from "../../../model/view/studentErrors";
+import * as React from 'react';
+import { Input, Select } from '../../../common/components/';
+import { StudentView } from '../../../model/view/studentView';
+import { StudentErrors } from '../../../model/view/studentErrors';
+import { CountryView } from '../../../model/view/countryView';
 
 interface Props {
   student: StudentView;
   errors: StudentErrors;
   fireFieldValueChanged: (viewModel: any, fieldName: string, value: any, filter?: any) => void;
```

Comment:
ViewModel could be extracted from state (if we thunk, action has got access to state but we could end up coupling the action to states), on the other hand the fieldName / Value / Filter could be encasuplated in a structure.